_ingest-pipelines/processors/dissect.md

```diff
@@ -0,0 +1,526 @@
+
+```bash
+New York, NY 10017
+New York City, NY 10017
+```
+
+The following example pipeline uses the right padding modifier with an empty key `%{->}`:
+
+```json
+PUT /_ingest/pipeline/dissect-test
+{
+  "description": "Pipeline that dissects web server logs",
+  "processors": [
+    {
+      "dissect": {
+        "field": "message",
+        "pattern": "[%{client_ip}]%{->}[%{timestamp}]" 
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+You can test the pipeline using the following example:
+
+```json
+POST _ingest/pipeline/dissect-test/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "message": "[192.168.1.10]   [03/Nov/2023:15:20:45 +0000]"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Your response should be similar to the following:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "client_ip": "192.168.1.10",
+          "message": "[192.168.1.10]   [03/Nov/2023:15:20:45 +0000]",
+          "timestamp": "03/Nov/2023:15:20:45 +0000"
+        },
+        "_ingest": {
+          "timestamp": "2024-01-22T22:55:42.090569297Z"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Append modifier (`+`)
+
+The append modifier combines the values of two or more values into a single output value. The values are appended from left to right. You can also specify an optional separator to be inserted between the values. 
+
+#### Example of usage
+
+The following is an example pipeline with an append modifier: 
+
+```json
+PUT /_ingest/pipeline/dissect-test
+{
+  "description": "Pipeline that dissects web server logs",
+  "processors": [
+    {
+      "dissect": {
+        "field": "message",
+        "pattern": "%{+address}, %{+address} %{+address}",
+        "append_separator": "|"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+You can test the pipeline using the following example:
+
+```json
+POST _ingest/pipeline/dissect-test/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "message": "New York, NY 10017"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+The substrings are appended to the `address` field, as shown in the following response:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "address": "New York|NY|10017",
+          "message": "New York, NY 10017"
+        },
+        "_ingest": {
+          "timestamp": "2024-01-22T22:30:54.516284637Z"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Append with order modifier (`+` and `/n`)
```

Comment:
Re: my previous comments, this modifier name is particularly awkward without some way of indicating that it's a name.